packages/grid/x-data-grid/src/hooks/features/clipboard/useGridClipboard.ts

```diff
@@ -81,16 +84,36 @@ export const useGridClipboard = (apiRef: React.MutableRefObject<GridPrivateApiCo
         return;
       }
 
-      apiRef.current.unstable_copySelectedRowsToClipboard();
+      let textToCopy = '';
+      const selectedRows = apiRef.current.getSelectedRows();
+      if (selectedRows.size > 0) {
+        textToCopy = apiRef.current.getDataAsCsv({
+          includeHeaders: false,
```

Comment:
One use-case could be to include headers for copy and paste, do you think it's worth considering introducing props like `unstable_includeHeadersDuringExport` and `unstable_includeHeadersDuringImport`? (perhaps in a follow-up PR)